### DISCUSSION TITLE
Support for multiple styles within a file

### DISCUSSION THREAD

**Discussion by VedAustin** [mkdocstrings/mkdocstrings#554] [Q&A] [answered]
Hello .. thanks for the wonderful package.
I have a repo at our org, where within a single python file, there are multiple docstring style formats: `sphinx` and `google`. For example:
```python
# some file.py
def foo(a: int):
"""
:param a:
"""
...


def bar(b: int):
"""
Args:
   b (int): 
"""
...
```

Is it possible to parse such files where there are multiple functions with different docstring style formats? 
TIA

**Comment by pawamoy** [comment, marked as answer]
Hi @VedAustin, glad you like the project!

Griffe's model supports having different styles in different docstrings, but with mkdocstrings it requires that you inject each function individually:

```md
::: some_file.foo
    options:
      docstring_style: sphinx

::: some_file.bar
    options:
      docstring_style: google
```

There's no way yet to auto-detect the style, but we previously discussed this possibility, see discussion here: https://github.com/mkdocstrings/griffe/issues/5.
